src/main/java/io/confluent/connect/elasticsearch/ElasticsearchClient.java

```diff
@@ -185,6 +187,9 @@ public boolean createIndex(String index) {
     if (indexExists(index)) {
       return false;
     }
+    if (config.isDataStream()) {
+      return createDataStream(index);
+    }
 
     CreateIndexRequest request = new CreateIndexRequest(index);
```

Comment:
how about we move this to another `private boolean createIndex(String index)` method and then have this method called `public boolean createIndexOrDataStream(String name)`

Reply:
IMO should be pretty self-explanatory, especially when reading the code